src/class/msc/msc_device.h

```diff
@@ -164,6 +182,12 @@ ATTR_WEAK void tud_msc_read10_complete_cb(uint8_t lun);
 ATTR_WEAK void tud_msc_write10_complete_cb(uint8_t lun);
 ATTR_WEAK void tud_msc_scsi_complete_cb(uint8_t lun, uint8_t const scsi_cmd[16]);
 
+// Hook to make a mass storage device read-only.
+ATTR_WEAK bool tud_msc_is_writable_cb(uint8_t lun);
+
+// Override for dynamic LUN sizes.
+ATTR_WEAK bool tud_lun_capacity_cb(uint8_t lun, uint32_t* last_valid_sector, uint16_t* block_size);

```

Comment:
yeah, capacity callback is better than using CFG_TUD_MSC_BLOCK_NUM/CFG_TUD_MSC_BLOCK_SZ for multiple LUN support. I will remove those macro in the config file and use this instead. Though the function name should be tud_msc_read_capacity_cb()